<launch>
	<!-- vim: ft=xml -->
	<!-- base node launch file-->

	<arg name="fcu_url" />
	<arg name="gcs_url" />
	<arg name="system_id"/>
	<arg name="tgt_system" />
	<arg name="tgt_component" />
	<arg name="blacklist_yaml" />
	<arg name="config_yaml" />
	<arg name="ref_msg"/>

	<node pkg="mavros" type="mavros_node" name="mavros" clear_params="true" output="screen" respawn="true">
		<param name="fcu_url" value="$(arg fcu_url)" />
		<param name="gcs_url" value="$(arg gcs_url)" />
		<param name="system_id" value="$(arg system_id)"/>
		<param name="target_system_id" value="$(arg tgt_system)" />
		<param name="target_component_id" value="$(arg tgt_component)" />
        <param name="conn_heartbeat" value="5" />
		<param name="conn_timeout" value="5.0" />
        <!-- automatically start mavlink on USB -->
		<param name="startup_px4_usb_quirk" value="true" />

		<!-- load blacklist, config -->
		<rosparam command="load" file="$(arg blacklist_yaml)" />
		<rosparam command="load" file="$(arg config_yaml)" />

		<remap from="/state/reference" to="$(arg ref_msg)"/>
		<remap from="/mavros/position/local_nwu/sys_id_11" to="/mavros/position/local_nwu/sys_id_$(arg system_id)"/>
	</node>

	<!--node pkg="mavros" type="gcs_bridge" name="mav_gcs_bridge" required="true" clear_params="true" output="screen">
		<param name="gcs_url" value="/dev/ttyUSB1:57600" />
	</node-->

</launch>
